pkg/common/volume_operations.go

```diff
@@ -424,6 +428,78 @@ func (vo *VolumeOperationsImpl) WaitStatus(ctx context.Context, volumeID string,
 	}
 }
 
+// ExpandVolume updates Volume status to Resizing to trigger expansion in reconcile, if volume has already had status
+// Resizing or Resized, function doesn't do anything. In case of statuses beside VolumeReady, Created, Published function return error
+// Receive golang context, volume CR, requiredBytes as int
+// Return volume spec, error
+func (vo *VolumeOperationsImpl) ExpandVolume(ctx context.Context, volume *volumecrd.Volume, requiredBytes int64) (*api.Volume, error) {
+	defer vo.metrics.EvaluateDurationForMethod("ExpandVolume")()
+	ll := vo.log.WithFields(logrus.Fields{
+		"method":   "ExpandVolume",
+		"volumeID": volume.Spec.Id,
+	})
+	currStatus := volume.Spec.CSIStatus
+	switch currStatus {
+	case apiV1.Resizing, apiV1.Resized:
+		ll.Debug("Volume is already expanding")
+	case apiV1.VolumeReady, apiV1.Created, apiV1.Published:
+		if !util.IsStorageClassLVG(volume.Spec.StorageClass) {
+			return nil, status.Error(codes.FailedPrecondition, "Volume doesn't have LVG storage class")
+		}
+		capacity, err := vo.crHelper.GetACByLocation(volume.Spec.Location)
+		if err != nil {
+			ll.Errorf("Failed to get AC by location %s", volume.Spec.Location)
+			return nil, status.Error(codes.Internal, "Unable to read AC")
+		}
+
+		if capacity.Spec.Size < requiredBytes {
+			return nil, status.Error(codes.OutOfRange, "Required bytes are less than existing LVG size")
+		}
+		volume.Spec.CSIStatus = apiV1.Resizing
```

Comment:
Shouldn't we reserve space in AC first and reject request right away if update AC failed?

Reply:
Yes, it makes sense